plugins/woocommerce-blocks/assets/js/data/cart/thunks.ts

```diff
@@ -39,18 +39,24 @@ export const receiveCart =
 		dispatch: CartDispatchFromMap;
 		select: CartSelectFromMap;
 	} ) => {
-		const newCart = camelCaseKeys( response ) as unknown as Cart;
+		const cartResponse = camelCaseKeys( response ) as unknown as Cart;
 		const oldCart = select.getCartData();
 		const oldCartErrors = [ ...oldCart.errors, ...select.getCartErrors() ];
 
-		updateCartErrorNotices( newCart.errors, oldCartErrors );
+		// Set data from the response.
+		dispatch.setCartData( cartResponse );
+
+		// Get the new cart data before showing updates.
+		const newCart = select.getCartData();
```

Comment:
This ensures the `newCart` is complete, because the data from the action may be incomplete.